File: Mongo-Developer-M101P-Pyramid/Week 2/HW2.1/answer.js
 //use simple query
 db.grades.find({type:"exam",score:{$gte:65}}).sort({score:1}).limit(1)


 //use aggregate framework
db.grades.aggregate([{$project:{type:1,score:1,student_id:1}},
	{'$match':{type:"exam",score:{$gte:65},student_id:{$exists:true}}},{$sort:{score:1}},{$limit:4}])

File: Mongo-Developer-M101P-Pyramid/Week 2/HW2.1/answer.py
import pymongo
from pymongo import MongoClient
import sys
import json
try:
	connection = MongoClient('localhost', 27017)
	db = connection.students
	grades = db.grades
	_query = {'type':'exam','score':{'$gte':65}}
	_datas = grades.find(_query).sort('score',pymongo.ASCENDING).limit(1)
	for _data in _datas:
		print _data
except:
	print sys.exc_info()
